x/staking/types/authz.go

```diff
@@ -1,21 +1,20 @@
 package types
 
 import (
-	context "context"
+	"context"
+	"fmt"
 
 	errorsmod "cosmossdk.io/errors"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/authz"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	"github.com/cosmos/cosmos-sdk/x/authz"
 )
 
 // TODO: Revisit this once we have propoer gas fee framework.
 // Tracking issues https://github.com/cosmos/cosmos-sdk/issues/9054, https://github.com/cosmos/cosmos-sdk/discussions/9072
 const gasCostPerIteration = uint64(10)
 
-var _ authz.Authorization = &StakeAuthorization{}
```

Comment:
This is quite handy, if we go this way, cannot we move the interface in `types/authz`?
Codec (`x/bank/types/codec.go`) still requires the interface registration anyway, so it does not really remove the cyclic dep as of now.

Reply:
>Codec (x/bank/types/codec.go) still requires the interface registration anyway, so it does not really remove the cyclic dep as of now.

Oops, I meant to move those registrations to authz.  Will do that now.

>This is quite handy, if we go this way, cannot we move the interface in types/authz?

This is handy while writing, but obviates one of the best features of how interfaces are implemented in go's type system.  A type can implement an interface's _behavior_ without forming an explicit relationship on the package in which it is specified.